### DISCUSSION TITLE
Adjust Popup position with margin

### DISCUSSION THREAD

**Discussion by Cfun1** [xamarin/XamarinCommunityToolkit#1188] [General]
Is there a way to adjust the popup position? for example I don't want it to be positioned at the extra bottom but with a slight vertical margin `VerticalOptions="End" HorizontalOptions="Center"` plus setting a margin to the popup itself. But , I don't see a `Margin` property.

**Comment by SkyeHoefling** [comment]
The Popup control doesn't support a position margin as of right now. The way I have accomplished this is using a transparent background color and then adding the margin to the content view of the popup.

**Reply by Cfun1** [reply]
I have tried to modify this file https://github.com/Cfun1/XamsungHealth/blob/master/XamsungHealth/Views/Popup/CancelPopup.xaml as you explained, by setting a margin on Frame or even added a main child as a Grid  but it doesn't seem to work, maybe be I am missing something.
> The Popup control doesn't support a position margin as of right now.

Is there any plan for a future enhancement ?

**Reply by SkyeHoefling** [reply]
I just peeked at your code in the `CancelPopup.xaml` you did the right thing by setting the `Popup.Color` to transparent. The `Frame` control doesn't appear to be configured correct and I would expect this to not work. You set the Padding which specifies the space between the `Frame` and the content inside of the `Frame`. If you want to leverage the transparent `Popup` you should really use `Margin` to specify the space between the bounds of your `Frame` and the bounds of your `Popup`. 

Try updating your Popup Code to look like this
```xml
<xct:Popup Color="Transparent">
  <Frame Margin="20" Padding="20">
    <!-- content goes here -->
  </Frame>
</xct:Popup>
```

If that doesn't work You can get creative with something like this
```xml
<Frame BackgroundColor="Transparent" Padding="20">
  <Frame>
    <!-- content goes here -->
  </Frame>
</Frame>
 ```

Using a double `Frame` should not be necessary but it should do the trick if the above approach doesn't work. I wrote both of these code snippets without testing so your results may not be exact.

@ahoefling
>The Popup control doesn't support a position margin as of right now.

@cfun1
> Is there any plan for a future enhancement ?

I do not have any plans to add a spec for margin position at the moment but I do see value in it. Would you be willing to put your thoughts into a full specification?

**Comment by Cfun1** [comment]
hey @ahoefling thanks for your answer, i have tried the first approach and it didn't work, the second one did but it kind of a workaround. I believe it would be useful to have the ability to adjust popup with a margin value.

[ [Spec] Popup add margin property #1398 ](https://github.com/xamarin/XamarinCommunityToolkit/issues/1398)
